### PR TITLE
Add -nr flag to exclude types from code-generation

### PR DESCRIPTION
**Pull request by 2opremio**

### CODE REVIEW

**Comment by 2opremio** [comment]
An even better option for me would be not generating `Selfer`s for types which already implement them. But it is more involved than this or #139 and unfortunately I don't have time to spend on it right now.

@ugorji Does it sound like a good option to you?

**Comment by ugorji** [comment]
@2opremio We don't generate Selfer for types which already implement them.
I think you are talking about #142 where we mistakenly overlook the CodecDecodeSelfer method for a type. Let me look into this quickly. For now, I will keep this -nr flag on hold since it is not urgent, and think about it more.

**Comment by 2opremio** [comment]
I am not talking about #142. In my experience codecgen does generate selfers for types that already implement then. I've had to disable each of them and that's what motivated this PR.

**Comment by ugorji** [comment]
Ah. verified. That is a bug.

The fix is simple, and I have it done. However, I am concerned about making this change, because it could cause problems for folks who depend on it.

Question: do you always delete the old codecgen output file, before running codecgen again. I would hate to fix it, and then folks run codecgen, and silently we do nothing for some types which already have this. The option would be to print a report when done, and/or exit with a non-zero exit if codecgen didn't run for all types.

Thoughts?

**Comment by 2opremio** [comment]
> Question: do you always delete the old codecgen output file, before running codecgen again.

I am not currently removing the file.

>  I would hate to fix it, and then folks run codecgen, and silently we do nothing for some types which already have this. 

I don't see how things can break. If the types already implement Selfer, generating Selfers would cause an error anyways.

**Comment by 2opremio** [comment]
> you can always hide this behavior under a flag.

Actually, I would do the opposite. Create a flag to generate selfers for types, even if they already implement them. In this way, even if you break something for users, they can always add a flag to get the old behavior.

File: codec/codecgen/gen.go [hunk 1]
```diff
@@ -180,6 +180,9 @@ func Generate(outfile, buildTag, codecPkgPath string, uid int64, useUnsafe bool,
 						//   FuncType, InterfaceType, StarExpr (ptr), etc
 						switch td.Type.(type) {
 						case *ast.StructType, *ast.Ident, *ast.MapType, *ast.ArrayType, *ast.ChanType:
+							if notRegexName.FindStringIndex(td.Name.Name) != nil {
```
Comment (ugorji) [inline codec/codecgen/gen.go]: Can you combine both of these into one line i.e. if notRegexName... == nil && regexName... != nil { __add_to_types __ }

Thanks.

**Comment by 2opremio** [comment]
I have addressed the comment. The AST loop in `Generate` is crying for a refactoring, it's close to reaching a cyclomatic complexity record.

**Pull request merged**